include/exadg/time_integration/bdf_time_integration.h

```diff
@@ -79,10 +94,15 @@ compute_bdf_time_derivative(VectorType &                       derivative,
                             BDFTimeIntegratorConstants const & bdf,
                             double const &                     time_step_size)
 {
-  derivative.equ(bdf.get_gamma0() / time_step_size, solution_np);
+  std::vector<VectorType const *> previous_solutions_ptrs(previous_solutions.size());
 
-  for(unsigned int i = 0; i < previous_solutions.size(); ++i)
-    derivative.add(-bdf.get_alpha(i) / time_step_size, previous_solutions[i]);
+  std::transform(previous_solutions.begin(),
+                 previous_solutions.end(),
+                 previous_solutions_ptrs.begin(),
+                 [](VectorType const & t) { return &t; });
```

Comment:
I wonder whether one could do the transform in the other direction to receive `std::vector<VectorType const &>` by looping over the ptrs vector and calling `previous_solution.push_back(...)`. I am thinking about this because I would prefer the function containing the actual implementation to use value-semantics. This would require that the assigment/type conversion `std::vector<VectorType const &> const & <-> std::vector<VectorType> const &` can be done automatically (I think so ...).

Reply:
It is not possible to have a `std::vector<T const&>`, see https://godbolt.org/z/8T4xx8Tdj. This is probably also the reason get_velocies_and_times returns `std::vector<VectorType const*>`.